lib/utils/address.ts

```diff
@@ -15,30 +55,143 @@ export function convertSS58Format(address: string, prefix: number): string {
 }
 
 /**
- * Replaces the last index in a BIP44 derivation path with a new index.
- * This modifies the address_index (5th component) in the path.
- *
+ * Parses a BIP44 derivation path into its components
+ * @param path - The BIP44 path to parse (e.g., "m/44'/354'/0'/0'/0'")
+ * @returns Parsed BIP44 components
+ * @throws {Bip44PathError} If the path is invalid or malformed
+ */
+export function parseBip44Path(path: string): Bip44Components {
+  if (!path || typeof path !== 'string') {
+    throw new Bip44PathError('BIP44 path must be a non-empty string', path)
+  }
+
+  // Remove any whitespace and validate basic format
+  const trimmedPath = path.trim()
+  if (!trimmedPath.startsWith('m/')) {
+    throw new Bip44PathError('BIP44 path must start with "m/"', path)
+  }
+
+  // Split the path and validate structure
+  const parts = trimmedPath.split('/')
+  if (parts.length !== 6) {
+    throw new Bip44PathError("BIP44 path must have exactly 6 components (m/purpose'/coinType'/account'/change'/addressIndex')", path)
+  }
+
+  // Validate and parse each component
+  const [m, purposeStr, coinTypeStr, accountStr, changeStr, addressIndexStr] = parts
+
+  if (m !== 'm') {
+    throw new Bip44PathError('BIP44 path must start with "m"', path, 'purpose')
+  }
+
+  // Helper function to parse hardened index
+  const parseHardenedIndex = (str: string, componentName: string): number => {
+    if (!str.endsWith("'")) {
+      throw new Bip44PathError(`BIP44 ${componentName} must be hardened (end with \')`, path, componentName)
+    }
+    const numStr = str.slice(0, -1)
+    const num = Number.parseInt(numStr, 10)
+    if (Number.isNaN(num) || num < 0) {
+      throw new Bip44PathError(`BIP44 ${componentName} must be a non-negative integer`, path, componentName)
+    }
+    return num
+  }
+
+  try {
+    const purpose = parseHardenedIndex(purposeStr, 'purpose')
+    const coinType = parseHardenedIndex(coinTypeStr, 'coinType')
+    const account = parseHardenedIndex(accountStr, 'account')
+    const change = parseHardenedIndex(changeStr, 'change')
+    const addressIndex = parseHardenedIndex(addressIndexStr, 'addressIndex')
```

Comment:
Any reason why we are forcing all parts to be hardened? This is not what wallets usually use to derive accounts. From experience, first three are forced to be hardened, but the last two not really. The derived address will be different.